scipy/_lib/_array_api.py

```diff
@@ -514,7 +515,7 @@ def xp_ravel(x: Array, /, *, xp: ModuleType | None = None) -> Array:
 
 
 # utility to broadcast arrays and promote to common dtype
-def xp_broadcast_promote(*args, ensure_writeable=False, force_floating=False, xp=None):
+def xp_broadcast_promote(*args, copy=False, force_floating=False, xp=None):
```

Comment:
This was misnamed from the time when args could only be numpy. a copy of a numpy read-only array is only writeable - which is not something you can say about JAX or PyData Sparse arrays.

The intent of this parameter is not to ensure that you can write to the arguments, but that you do not accidentally write back to the inputs.
I've renamed the parameter to match the same in `astype`.

I'll add documentation once conflict with https://github.com/scipy/scipy/pull/22735 is resolved.